util/opentelemetry-util-http/src/opentelemetry/util/http/__init__.py

```diff
@@ -321,6 +340,7 @@ def detect_synthetic_user_agent(user_agent: Optional[str]) -> Optional[str]:
 
     Note: Test patterns take priority over bot patterns.
     """
+    user_agent = normalize_user_agent(user_agent)
```

Comment:
In all the instrumentations you have updated you have the normalized version around, maybe keep the signature as `str | None` and pass the normalized one? I know we don't have type checking yet there but still we won't be able to revert the signature later.